[Fix]#45 customAlertView, 그만하기, 일시정지 화면 UI구현 및 세부 로직 수정

Sophie4han opened the pull request:
close #45 
<!-- close #No(이슈 넘버 등록하면 자동으로 이슈가 닫힙니다. 이슈를 닫고 싶지 않다면, 즉 해당 이슈의 작업이 아직 끝나지 않았다면 닫지 않습니다.) -->

## *⛳️ Work Description*

- customAlertView 구현
- 일시정지, 그만하기 화면 구현

## *📸 Screenshot*

<img src ="https://github.com/user-attachments/assets/d124a23a-be84-4132-8a68-ea1a98fd4eda" width = 50%>

<img src ="https://github.com/user-attachments/assets/7c1f7274-d1e9-4ba4-b1e9-fa4937b2ba65" width = 50%>


## *📢 To Reviewers*

<!-- 이 PR을 리뷰하는 사람들에게 할 말이 있다면 작성하시면 됩니다. -->

- 이슈 4번은 무시해주세요
- 닉네임 등록 및 코드 공유, 상대방 코드 입력 화면은 무시해주세요

## *✅ Checklist*

<!-- PR을 올리기 전에 PR을 올리는 대상자가 점검할 내용입니다. -->

- [x]  주석 및 프린트문 제거 확인.
- [x]  컨벤션 준수 확인.

Sophie4han committed:
[Design]#4-LoginViewController UI 작업 1차

Sophie4han committed:
[Design]#4-LoginViewController UI 작업 1차

Sophie4han committed:
[Design]#4 - LoginViewController UI 구현 완료, Asset 로고 및 체크박스 이미지 추가

Sophie4han committed:
[Design]#4 - LoginViewController UI 구현 완료, Asset 로고 및 체크박스 이미지 추가

Sophie4han committed:
[Design] signUp UI 1차

Sophie4han committed:
[Design] signUp UI 1차

Sophie4han committed:
Merge branch 'dev' of https://github.com/tututu08/FitMate into Design/#4

Sophie4han committed:
[Chore] .gitignore 적용

Sophie4han committed:
[Design] Nickname Register UI 구현 완료

Sophie4han committed:
[Design] 코드 공유 화면 UI 구현 완료

Sophie4han committed:
[Design] 상대방 코드 입력 화면 UI 구현 완료

Sophie4han committed:
[Design] 상대방 코드 입력 화면 UI 구현 완료

Sophie4han committed:
[Feat] 닉네임 등록뷰 및 코드 기입뷰 간 화면 전환 로직 추가

Sophie4han committed:
[Design] CustomAlertVIew 구현 완료 및 텍스트 필드 eyeToggleButton 삭제

Sophie4han committed:
[Design] CustomAlertView UI 구현완료

Sophie4han committed:
[Design] 일시정지 alert UI 화면 구현

Sophie4han committed:
[Design] 일시정지 alert UI 화면 구현

Sophie4han committed:
[Design] 일시정지 알림창 UI 화면 구현2

Sophie4han committed:
[Design] 일시정지 화면 하단 버튼 수정

Sophie4han committed:
[Design] 그만하기 화면 UI 구현 및 일시정지 화면 버튼명 수정

Commit:
Merge branch 'Design/#25' of https://github.com/tututu08/FitMate into Fix/#45

# Conflicts:
#	FitMate/FitMate/App/SceneDelegate.swift

Commit:
Merge branch 'Design/#26' of https://github.com/tututu08/FitMate into Fix/#45

# Conflicts:
#	FitMate/FitMate/Resources/Assets.xcassets/pause.imageset/Contents.json
#	FitMate/FitMate/Resources/Assets.xcassets/stop.imageset/Contents.json

Commit:
Merge branch 'Design/#27' of https://github.com/tututu08/FitMate into Fix/#45

Commit:
[Fix] 커스텀알림창 및 일시정지, 그만하기 화면 수정

Commit:
[Fix] 커스텀알림창 및 일시정지, 그만하기 화면 수정

Commit:
[Chore] 구글 인포 삭제

mnh4140 commented on FitMate/FitMate/Common/Base/CustomAlertView.swift:
```diff
@@ -0,0 +1,145 @@
+        private var resumeButton: UIButton?
+        private var stopButton: UIButton?
+        
+        func setIcon(_ icon: UIImageView) -> AlertBuilder {
+            self.icon = icon
+            /// 함수 자체를 chaining
+            /// 다음 호출을 이어가기 위한 객체(self) 자체를 반환해야지
+            /// icon을 리턴하면 다른 요소들 설정 메서드를 이어서 못씀
+            /// return self -> builder pattern의 핵심 문법
+            return self
+        }
```
builder pattern 신기하네요.

자신을 리턴하면 어떻게 되는 건가요?

mnh4140 replied:
여기서 self 는 누구인가요?

Sophie4han replied:
self는 현재 AlertBuilder 인스턴스 자신입니다
return self를 함으로써 메서드 체이닝이 가능해집니다
음 .setIcon().setTitle().setMessage()와 같은 설정 메서드의 체이닝을 위해 스스로를 리턴한다고 생각하시면 됩니담

mnh4140 replied:
메서드 체이닝도 설명해주세요

mnh4140 commented on FitMate/FitMate/Common/Base/CustomAlertView.swift:
```diff
@@ -0,0 +1,145 @@
+                return view
+            }()
+            
+            alertView.snp.makeConstraints { make in
+                make.height.equalTo(349)
+                make.width.equalTo(326)
+            }
```
제약설정은 따로 메소드로 빼도 되지 않을까요?

Sophie4han replied:
참고하도록 하겠습니다

h-yoon replied:
저도 노훈님과 같은 생각입니다. buildAlert()의 뷰 구성과 레이아웃 설정을 섞은것이라 책임분리가 안되어있는 것 같습니다.
private func applyLayout(to alertView: UIView) 같은 형식으로 별도 메서드로 빼는것이 좋을 것 같습니다!

mnh4140 commented on FitMate/FitMate/Common/Base/PauseAlertViewController.swift:
```diff
@@ -0,0 +1,42 @@
+        view.backgroundColor = UIColor.black.withAlphaComponent(0.5)
+        
+        let alert = CustomAlertView.AlertBuilder()
+            .setIcon(UIImageView(image: UIImage(named: "pause")))
+            .setTitle("운동이 잠시 멈췄어요")
+            .setMessage("운동이 일시정지 되었습니다. \n준비되면 이어서 계속해 보세요!")
+            .setResumeButton(resumeButton)
+            .buildAlert()
```
self 반환을 하면 이렇게 쓰일수있는 건가요?

Sophie4han replied:
네 맞습니다
설명 전에 스스로 이해하셨군요
굿

mnh4140 commented on FitMate/FitMate/Common/Base/PauseAlertViewController.swift:
```diff
@@ -0,0 +1,42 @@
+            .setResumeButton(resumeButton)
+            .buildAlert()
+        
+        view.addSubview(alert)
+        alert.snp.makeConstraints { make in
+            make.center.equalToSuperview()
+        }
+    }
```
viewDidLoad에 메서드로 정의되지 않은 코드들이 많은것 같습니다.

각 코드의 용도별로 메서드를 나눠서 정의하고 호출하여 쓰는것이 가독성에 좋을 것 같습니다.

Sophie4han replied:
넵 반영하여 수정하도록 하겠습니다

mnh4140 commented on FitMate/FitMate/Common/Base/StopAlertViewController.swift:
```diff
@@ -0,0 +1,53 @@
+        return stop
+    }()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        view.backgroundColor = UIColor.black.withAlphaComponent(0.5)
+        
+        let alert = CustomAlertView.AlertBuilder()
+            .setIcon(UIImageView(image: UIImage(named: "stop")))
+            .setTitle("정말 그만하시겠어요?")
+            .setMessage("기록은 안전하게 저장됩니다. \n단, 이 선택은 상대방의 운동도 함께 중단시킵니다. \n상대방도 준비가 되었는지 확인해 주세요.")
+            .setResumeButton(resumeButton)
+            .setStopButton(stopButton)
+            .buildAlert()
+       
+        view.addSubview(alert)
+        alert.snp.makeConstraints { make in
+            make.center.equalToSuperview()
+        }
+    }
```
메서드로 분리...

Sophie4han replied:
네..

mnh4140 reviewed (approved):
코드에 주석이 추가되면 좋을 것 같습니다.

고생하셨습니다!

brody424 commented:
해당 프로젝트 빌드 되나요..?
저는 빌드오류 발생해요 ㅠ

<img width="1125" alt="스크린샷 2025-06-11 오후 4 54 11" src="https://github.com/user-attachments/assets/419dd961-b8fa-446a-9e58-a8a8b26effd6" />

Kimewg reviewed (approved):
수고하셨습니도

h-yoon commented on FitMate/FitMate/Common/Base/CustomAlertView.swift:
```diff
@@ -0,0 +1,145 @@
+        }
+        
+        // alert 본체 set하는 메서드
+        func buildAlert() -> UIView {
```
현재 buildAlert()는 UIView를 반환하고 있는데, 이 클래프 자체가 CustomAlertView라서 자기 자신을 설정하는 방향도 고려해볼만 한 것 같습니다! 그리고 커스텀 AlertView 타입을 리턴하도록 하면 타입 일관성이 높아질 것 같습니다!

tututu08 replied:
저도 좋은 생각 같습니다. 그러면 정말 커스텀알럿의 기능이 더 높아질것 같네요!

h-yoon commented on FitMate/FitMate/Common/Base/CustomAlertView.swift:
```diff
@@ -0,0 +1,145 @@
+            icon.snp.makeConstraints { make in
+                make.top.equalToSuperview().inset(32)
+                make.leading.trailing.equalToSuperview().inset(120)
+                make.height.equalTo(icon.snp.width).multipliedBy(1.0)
```
1.0 은 불필요해보입니다. make.height.equalTo(icon.snp.width)로 간결하게 표현하셔도 좋을 것 같습니다!

tututu08 replied:
아 그래도 정사각형이되겠네요

h-yoon commented on FitMate/FitMate/Common/Base/CustomAlertView.swift:
```diff
@@ -0,0 +1,145 @@
+    private let buttonStack = UIStackView()
+    private let background = UIView() // alert 뒤에 화면
+    
+    override init(frame: CGRect) {
```
현재 CustomAlertView안에 실질적으로 아무 역할도 안하고있는 것 같아요 실질적 뷰구성이 모두 buildAlert안에있네용 ㅜ

h-yoon commented on FitMate/FitMate/Scene/Login/View/CodeShareVIewController.swift:
```diff
@@ -0,0 +1,109 @@
+        defaultAvatarImage.snp.makeConstraints { make in
+            make.top.equalTo(guideMent.snp.bottom).offset(20)
+            make.leading.trailing.equalToSuperview().inset(110)
+            make.height.equalTo(defaultAvatarImage.snp.width).multipliedBy(1.0)
```
위의 내용과 같습니다.

h-yoon commented on FitMate/FitMate/Scene/Login/View/CustomTextField.swift:
```diff
@@ -0,0 +1,65 @@
+        return bounds.inset(by: UIEdgeInsets(top: 0, left: 12, bottom: 0, right: 0))
+    }
+    
+    func bind() {
```
바인드 메서드는 외부에서 호출하지 않으니 private 쓰는게 나아보여요

h-yoon commented on FitMate/FitMate/Scene/Login/View/MateCodeViewController.swift:
```diff
@@ -0,0 +1,113 @@
+    private let backButton: UIButton = {
+        let back = UIButton()
+        back.setImage(UIImage(named: "backButton"), for: .normal)
+        back.setTitle("              메이트 코드 입력", for: .normal)
```
이거 띄어쓰기로 정렬하신건가여? 오타면 수정 부탁드려요
만약 띄워쓰기 하려고 하신거면 디바이스환경에 따라 깨질 수 있어서 UIButton 자체에 titleLabel하고 imageView의 inset설정을 하는 방식이 더 안전해보입니다.

Sophie4han replied:
해당 내용 이전 PR에서 수정되었습니다!

h-yoon commented on FitMate/FitMate/Scene/Login/View/MateCodeViewController.swift:
```diff
@@ -0,0 +1,113 @@
+        completeButton.snp.makeConstraints { make in
+            make.top.equalTo(fillInMateCode.snp.bottom).offset(20)
+            make.leading.trailing.equalTo(view.safeAreaLayoutGuide).inset(20)
+            make.width.equalTo(335)
```
코드로 봤을 땐 이미 위에 leading~~.inset(20)으로 설정이 되어있어서 따로 width를 사용할 필요가 없어보이는데 꼭 있어야하는 코드인가여?

h-yoon commented on FitMate/FitMate/Scene/Login/View/NIcknameViewController.swift:
```diff
@@ -0,0 +1,135 @@
+        checkBoxImage.snp.makeConstraints { $0.size.equalTo(24) }
+        
+        let privacyTitle = UILabel()
+        privacyTitle.text = "이용약관(필수)"
```
이거 보통 개인정보처리방침(필수)라고 적지 않나여????

Sophie4han replied:
일단 피그마에 적힌 대로 작업해서 문구도 그대로 들어갔습니다!
수정하도록 하겠습니다!

h-yoon commented on FitMate/FitMate/Scene/Login/View/NIcknameViewController.swift:
```diff
@@ -0,0 +1,135 @@
+        registerButton.snp.makeConstraints { make in
+            make.bottom.equalTo(view.safeAreaLayoutGuide).inset(32)
+            make.leading.trailing.equalTo(view.safeAreaLayoutGuide).inset(20)
+            make.width.equalTo(335)
```
이것도 위와 동일합니다. 한번 확인해주세여!

h-yoon reviewed (approved):
고생하셨습니다

tututu08 commented on FitMate/FitMate/Common/Base/CustomAlertView.swift:
```diff
@@ -0,0 +1,145 @@
+
+class CustomAlertView: UIView {
+
+    private let containerVIew = UIView() // alert 본체
```
containerVIew -> containerView 수정 부탁드립니다!

Commit:
Merge branch 'dev' of https://github.com/tututu08/FitMate into Fix/#45

# Conflicts:
#	FitMate/FitMate/Resources/Assets.xcassets/image/stop.imageset/stop.png
#	FitMate/FitMate/Resources/Assets.xcassets/image/stop.imageset/Ｂutton (1).png
#	FitMate/FitMate/Scene/Login/View/LoginViewController.swift
#	FitMate/FitMate/Scene/Login/View/NIcknameViewController.swift

Commit:
[Chore] #45 메서드 분리 및 CustomAlertView 재사용 가능하게 수정

h-yoon committed:
[Fix]#62 이슈36 + dev브랜치 중복파일 삭제 (#63)

* Fix: 이슈36 + dev브랜치 중복파일 삭제 #62

* fix: 수정 #62

* fix: 수정2 #62

* [Chore] #62 레이아웃 수정

tututu08 committed:
[Merge] #50 대결/협력 줄넘기 모드 카운팅 집계 및 UI 바인딩 로직 구현(#41 업데이트) (#61)

* [Design] #7 협력모드 공통 UI View 생성

* [Design] #7 대결모드 공통 UI View 생성

* [Design] #24 BattleSportsView 위치 이동(Common/Components)

* [Chore] #24 테스트용 코드 삭제

* [Design] #24 라벨 위치 수정

* [Chore] #24 required init 및 네이밍 컨벤션 수정

* [Chore] #7 테스트용 코드 삭제

* [Refactor] #7 BaseViewController 구조 수정

* [Chore] #24 스택뷰 컨벤션 맞게 수정

* [Feat] #41 CooperationSportsView 갱신 기능 추가, 주석 설명 추가

* [Feat] #41 JumpRope 협력 UI/점프 카운트/진행률 ViewModel 바인딩

* [Chore] #41 패키지 추가 (Firebase, RxCoreLocation 등)

* [Chore] #41 BaseView 생성

* [Feat] #50 JumpRope 협력모드 UI 및 MVVM 구조 개선

- CooperationSportsView 레이아웃/진행률 UI 개선
- JumpRopeCoopViewController에서 Input/Output MVVM 패턴 반영 및 바인딩 처리
- 목표치/캐릭터 전달 로직 추가 및 테스트 데이터 반영

* [Chore] #50 BaseView, BaseViewController수정

* [Feat] #50 JumpRope 대결모드 UI 및 MVVM 구조 개선

- BattleSportsView 레이아웃/진행률 UI 개선(게이지바 두개 구분, 색 차이)
- JumpRopeBattleViewController에서 Input/Output MVVM 패턴 반영 및 바인딩 처리
- 목표치/캐릭터 전달 로직 추가 및 테스트 데이터 반영

* [Merge]

* [Chore] #50 #Dev pull이후. PR전 최신화

* [Chore] #50 Assets 이미지 중복 제거 및 디폴트 방향 설정(오른쪽)

* [Feat] #50 이미지 좌우반전 추가(운동화면)

* [Chore] #50 누락된 뷰 추가(NicknameViewController)

* [Chore] #50 이미지 파일 정리

h-yoon committed:
[Design] #66 마이페이지 설정창 UI 구현 (#68)

* [Design] #66 마이페이지 설정창 UI 구현

* [Design] #66 마이페이지 설정창 UI 구현

* [Design] #66 마이페이지 설정창 UI 구현

Sophie4han committed:
[Design] #29 로그인 화면 변경된 UI로 수정 (#64)

* [Design]#4-LoginViewController UI 작업 1차

* [Design]#4 - LoginViewController UI 구현 완료, Asset 로고 및 체크박스 이미지 추가

* [Design] 로그인화면 UI 구현 완료 및 베이스뷰 setLayoutUI() 추가

* [Chore] 구글 인포 삭제

* [Chore] 구글 인포 삭제2

* [Chore] 구글 인포 삭제 완료

* [Chore] 로그인 화면 수정된 UI 구현

---------

Co-authored-by: Sophie4han <hanshirley775@gamil.com>

Sophie4han committed:
[Design] #51 메인 화면 UI 구현 및 메이트 유무에 따른 분기처리 및 메인에 필요한 이미지 에셋에 추가 (#52)

* [Design] 메인 화면 UI 구현 및 메이트 유무에 따른 분기처리 및 메인에 필요한 이미지 에셋에 추가

* [Chore] Lottie 스테이지

* [Chore] 임시 탭바 삭제 및 제약 수정 및 베이스뷰 적용

---------

Co-authored-by: Sophie4han <hanshirley775@gamil.com>
